### DISCUSSION TITLE
RuntimeError: Torch is not able to use GPU; add --skip-torch-cuda-test to COMMANDLINE_ARGS on AMD GPU

### DISCUSSION THREAD

**Discussion by vKireina** [lshqqytiger/stable-diffusion-webui-directml#456] [Q&A] [answered]
Context: I was originally using version 1.6.0 of Automatic1111 just fine. Decided should update since there has been many. Updated to latest version and now get this error. (Yes I followed the AMD Install directions).

![Screenshot 2024-04-30 125302](https://github.com/AUTOMATIC1111/stable-diffusion-webui/assets/59129997/45cf83fd-731e-46e7-9a01-c62fd2399127)

I was also previously using version 3.10.11 of Python from the Microsoft Store which worked fine before. I did already try installing 3.10.6 and made no difference.

Specs:
GPU - AMD Radeon RX 6900 XT
CPU - AMD Ryzen 9 5900 12-Core Processor - 12 Cores
RAM - 32 GB
VRAM - 16368 MB - GDDR6 2000 MHz
Memory Type - GDDR6
Driver Version - 23.40.31.05-240417a-402338C-AMD-Software-Adrenalin-Edition
AMD Windows Driver Version - 31.0.24031.5001
AMD Audio Driver Version - 10.0.1.30
Windows Edition - Windows 11 Core (64 bit)
Windows Version - 23H2

I have already tried deleting everything and starting from scratch, I have tried adding arguments, I have tried downgrading back to 1.6.0, I have even tried the SD Forge webui which supposedly doesnt matter what GPU you have. All result in same error.

I followed these following video guides exactly and did a fresh install from scratch for each one I attempted in this order:

1. https://youtu.be/eO88i8o-BoY?si=rEZ1EzVuDrMT1e2G
2. https://youtu.be/8xR0vms0e0U?si=XiLYcUQZbEj8kctc
3. https://youtu.be/TJ98hAIN5io?si=IzJRhSjmvz59PZxG
4. https://youtu.be/mHvnHA04sbc?si=SsProDPebDvIbfDX
5. https://youtu.be/PdkBmDoUIy8?si=I7Tk3w8JmliqzzQp

Im seriously at a lost of what to do now. No matter what I do it just doesnt work despite following all these videos and guides meant specifically for AMD users. Any help would be appreciated.

**Comment by lshqqytiger** [comment, marked as answer]
Remove `venv` folder, and launch webui with `--use-directml`

**Reply by vKireina** [reply]
Well, surprisingly that worked! Which is strange considering the first video I followed with a fresh install had me enter in the following Args: --use-directml --opt-sub-quad-attention --no-half --disable-nan-check --autolaunch, which includes that one. Could the other args messed it up somehow? Either way you've saved me from hair loss with a very simple answer, thankyou.

**Reply by lshqqytiger** [reply]
But consider migrating to `--use-zluda`.
Guide for SD.Next (but will work for this project either): https://github.com/vladmandic/automatic/wiki/ZLUDA
Comparison: https://github.com/vladmandic/automatic/wiki/ZLUDA#comparison-directml

**Reply by vKireina** [reply]
Thankyou, I'll take a look at it. o7
